### ISSUE TITLE
Error using multiple parameters suffixed with an asterisk - works with v2.4

### ISSUE THREAD

**Issue by blikblum** [pillarjs/path-to-regexp#185]
Having a path with multiple params suffixed with an asterisk used to work up to version 2.4

In version 3.0 it matches only the first param.

Example:

```javascript
let paramNames = [];
let re = pathToRegexp("/files/:path*.:ext*", paramNames);
let match = "/files/my/photo.jpg/gif".match(re);
```

In v2.4 this is the matched params:

path: my/photo 
ext: jpg/gif

In v3.0:

path: my/photo.jpg/gif 
ext: undefined

Here's live example for v2.4:
https://codesandbox.io/s/x75l707pxw
and v3:
https://codesandbox.io/s/x209j3nkko

Extracted from a real project [test case](https://github.com/QubitProducts/cherrytree/blob/master/tests/unit/pathTest.js#L43), that fails when upgrading to v3.0

**Comment by blakeembrey** [comment]
@blikblum To be honest, it looks like 2.4 exhibited buggy behavior - I would not have expected that match. What do you expect is the correct behavior? 3.0 was released to fix this specific issue by the way, see https://github.com/pillarjs/path-to-regexp/blob/master/History.md#300--2019-01-13.

You can get the old behavior back by explicitly escaping the period, e.g. `"/files/:path*\\.:ext*"`.

**Comment by blikblum** [comment]
I don't have strong opinion of what is the correct behavior. 

Posted the issue to make it it aware (if not yet)

Given the previous behavior seems buggy, it can be closed and i will adjust the test accordingly

**Comment by blakeembrey** [comment]
Thanks @blikblum. I do believe it's a manifestation of the issue I resolved in 3.0.
